Sources/Imaging/Core/ImageReslice/index.js

```diff
@@ -32,68 +31,45 @@ function vtkImageReslice(publicAPI, model) {
   let optimizedTransform = null;
 
   function getImageResliceSlabTrap(tmpPtr, inComponents, sampleCount, f) {
-    let subTmpPtr = tmpPtr;
-    let m = inComponents;
     const n = sampleCount - 1;
-    do {
-      let result = subTmpPtr[0] * 0.5;
-      for (let k = n - 1; k !== 0; --k) {
-        subTmpPtr = tmpPtr.subarray(inComponents);
-        result += subTmpPtr[0];
+    for (let i = 0; i < inComponents; i += 1) {
+      let result = tmpPtr[i] * 0.5;
+      for (let j = 1; j < n; j += 1) {
+        result += tmpPtr[i + j];
       }
-      subTmpPtr = tmpPtr.subarray(inComponents);
-      result += subTmpPtr[0] * 0.5;
-      tmpPtr.fill(result * f, 0, 1);
-      tmpPtr = tmpPtr.subarray(1);
-    } while (--m);
+      result += tmpPtr[i + n] * 0.5;
+      tmpPtr[i] = result * f;
+    }
   }
 
   function getImageResliceSlabSum(tmpPtr, inComponents, sampleCount, f) {
-    let subTmpPtr = tmpPtr;
-    let m = inComponents;
-    const n = sampleCount - 1;
-    do {
-      let result = subTmpPtr[0];
-      let k = n;
-      do {
-        subTmpPtr = tmpPtr.subarray(inComponents);
-        result += subTmpPtr[0];
-      } while (--k);
-      tmpPtr.fill(result * f, 0, 1);
-      tmpPtr = tmpPtr.subarray(1);
-    } while (--m);
+    for (let i = 0; i < inComponents; i += 1) {
+      let result = tmpPtr[i];
+      for (let j = 1; j < sampleCount; j += 1) {
```

Comment:
what happened to `--n` ?

Reply:
Can I see why --n need to be?

Reply:
nevermind, you are right, it is not needed.